common/src/chain/tokens/mod.rs

```diff
@@ -13,11 +13,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use serialization::{Decode, Encode};
+use serialization::{Decode, DecodeAll, Encode};
 
 pub type TokenId = H256;
```

Comment:
This can be a problem. Please do like PoolId and DelegationId and create an ID type. Using a type alias can create an issue with addresses because it removes type safety completely.

Reply:
fixed